OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AreaAdminControllerTests.cs

```diff
@@ -250,6 +252,43 @@ public async Task Update_WithErrorResponse_ReturnsStatusCodeResult()
         Assert.That(result, Is.InstanceOf<StatusCodeResult>());
     }
 
+    [Test]
+    public async Task Update_WhenServiceThrowDbUpdateConcurrencyException_ReturnsBadRequest()
+    {
+        // Arrange
+        var updateAreaAdminDto = new AreaAdminDto();
+
+        areaAdminServiceMock
+            .Setup(x => x.UpdateAreaAdminAsync(It.IsAny<string>(), updateAreaAdminDto, It.IsAny<string>()))
+            .Throws<DbUpdateConcurrencyException>();
+
+        // Act
+        var result = await areaAdminController.Update(updateAreaAdminDto);
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result, Is.InstanceOf<BadRequestObjectResult>());
+    }
+
+    [Test]
+    public async Task Update_WhenUserIsNotAdmin_ReturnsForbidenResponse()
+    {
+        // Arrange
+        var updateAreaAdminDto = new AreaAdminDto { Id = string.Empty };
+
+        areaAdminServiceMock
```

Comment:
(Test logic?) Test says "_WhenUserIsNotAdmin_", but you're mocking db exception. It looks like previous test dublicate.

Reply:
It is dublicate...
The execution will not reach that point...
Evening, inattentive, I'll fix it